### PR TITLE
Improve database feedback

### PR DESCRIPTION
**Pull request by Robert-Krajcik**
A PR for @marlycormar to test

### CODE REVIEW

File: Utils/utils.R [hunk 1]
```diff
@@ -40,31 +40,69 @@ create_db <- function(){
...
+    tryCatch({
+      rs <- dbSendQuery(con, query)
+    }, error = function(err) {
+      message <- paste("dbSendQuery returned",err)
+      message(message, .loggit = FALSE)
+      loggit("ERROR", message)
+      dbDisconnect(con)
+    })
+    
+    dbClearResult(rs)
```
Comment (marlycormar) [inline Utils/utils.R:53]: Hi @Robert-Krajcik, when I run the app after removing the `database.sqlite` and the `loggit.json` files, I get the following error:

`dbSendQuery returned Error in dbSendQuery(con, query): object 'query' not found`

followed by 

` Error in dbClearResult(rs) : object 'rs' not found `

**Comment by Robert-Krajcik** [comment]
Needed to use `dbSendStatement` instead of `dbSendQuery` and use x instead of query in function.
Also issue message "No rows were returned from db_fun query" if and only if the package name is not 'Select'

File: Utils/utils.R (hunk 1 above)
Comment (Robert-Krajcik) [reply Utils/utils.R:55]: OK I believe I have this figured out, @marlycormar

**Comment by marlycormar** [comment]
Ready for my review.

File: Utils/utils.R [hunk 2]
```diff
@@ -40,31 +40,73 @@ create_db <- function(){
...
+      dbDisconnect(con)
+      errFlag <<- TRUE
+    },
+    finally ={
+      if (errFlag) return(NULL) 
+    })
+  
+  dat <- dbFetch(rs)
+  dbClearResult(rs)
+  if (nrow(dat) == 0 & str_detect(strng, "'Select'") == FALSE) {
```
Comment (marlycormar) [inline Utils/utils.R:87]: Hi @Robert-Krajcik , I get the following error when running the app on this branch:

`Warning: Error in str_detect: object 'strng' not found`

Comment (Robert-Krajcik) [reply Utils/utils.R:87]: OK I'll take a look in a few minutes

**Comment by Robert-Krajcik** [comment]
Hi @marlycormar 
Try it now.
The "strng" should have been "query"

**Comment by marlycormar** [comment]
Hey @Robert-Krajcik! Could you share a summary of this PR?

**Comment by Robert-Krajcik** [comment]
Sure, @marlycormar 
I'm planning on spending most of the day tomorrow on risk_assessment

**Comment by Robert-Krajcik** [comment]
The purpose of this PR is to replace un-helpful database warning and error messages
with more descriptive messages using `tryCatch()`

for example, a message text is created
`message <- paste0("db_fun warning:\n",query,"\nresulted in\n",warn)`
where this message shows the actual query that caused the warning.
This gets logged to `loggit.json`

error messages are handled in the same way
`message <- paste0("db_fun error:\n",query,"\nresulted in\n",err)`

For queries  using the `db_fun()` function
an informational message will be issued when no rows were returned by the function. 

For delete, update or insert commands using the `db_ins()` function
a descriptive error message will be displayed,
as well as the message
`zero rows were affected by db_ins() function.`
if no rows were affected.

**Comment by marlycormar** [comment]
I was able to test the db error (by deleting the db while the app was running) but not the db warning. Do you have any tips @Robert-Krajcik?

**Comment by Robert-Krajcik** [comment]
Let me search for something that generates a warning

**Comment by Robert-Krajcik** [comment]
Is the message
`No rows were returned from db_fun query`
helpful at all?
I'm thinking about commenting it out/removing it

**Pull request merged**